tpm2tools/signer.go

```diff
@@ -46,7 +48,13 @@ func (signer *tpmSigner) Sign(_ io.Reader, digest []byte, opts crypto.SignerOpts
 	if err != nil {
 		return nil, err
 	}
-	return sig.RSA.Signature, nil
+
+	if sig.RSA != nil {
+		return sig.RSA.Signature, nil
+	} else {
+		sigStruct := struct{ R, S *big.Int }{sig.ECC.R, sig.ECC.S}
+		return asn1.Marshal(sigStruct)
+	}
```

Comment:
This should be a switch statement on `sig.Alg`. Doing one of:

- `AlgRSASSA` => `return sig.RSA.Signature, nil`
- `AlgECDSA` => `return asn1.Marshal(struct{ R, S *big.Int }{sig.ECC.R, sig.ECC.S})`
- panic